harness/determined/pytorch/_reducer.py

```diff
@@ -184,7 +185,17 @@ def per_slot_reduce(self) -> Any:
         return self.values
 
     def cross_slot_reduce(self, per_slot_metrics: List) -> Any:
-        flat_metrics = [item for sublist in per_slot_metrics for item in sublist]
+        """
+        Call the reducer function on metrics in their original order.
+
+        Interleave metrics from different slots as we flatten
+        the list of metrics to undo the effect of sharding.
+        Note: this will only reconstruct the original order if the user
+        calls ``update()`` the same number of times per batch
```

Comment:
wait, this will only reconstruct the original order if the user calls `update()` exactly 1 time per batch, right?  If they call it 2 times every batch, the order would be... well I don't know that they would care about ordering in that case.